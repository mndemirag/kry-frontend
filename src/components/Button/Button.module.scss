@import "../../utils";

.button {
  cursor: pointer;
  text-transform: uppercase;
  height: 42px;
  font-weight: bold;
  font-size: 14px;
  text-shadow: none;
  min-width: 110px;
  border-radius: 7px;
  line-height: 13px;
  color: white;
  background-color: $kry-blue;
  border: none;
  margin: 10px;
}
