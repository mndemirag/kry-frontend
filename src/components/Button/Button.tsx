import { ButtonProps } from "../../types";

import styles from "./Button.module.scss";

export const Button: React.FC<ButtonProps> = ({ children, onClick, type }) => {
  return (
    <button
      className={styles.button}
      aria-label={children}
      type={type}
      onClick={() => onClick && onClick()}
    >
      {children}
    </button>
  );
};
