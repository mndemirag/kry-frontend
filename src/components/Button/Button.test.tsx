import { render } from "@testing-library/react";

import { Button } from "./Button";

describe("Button", () => {
  it("renders properly", () => {
    const { container } = render(<Button> Button </Button>);
    expect(container).toMatchSnapshot();
  });
});
