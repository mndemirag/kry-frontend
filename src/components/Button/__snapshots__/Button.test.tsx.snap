// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders properly 1`] = `
<div>
  <button
    aria-label=" Button "
    class="button"
  >
     Button 
  </button>
</div>
`;
